http/oidc/src/test/java/org/wildfly/security/http/oidc/OidcProviderMetadataTest.java

```diff
@@ -0,0 +1,61 @@
+package org.wildfly.security.http.oidc;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test OIDC json config class properly returns boolean values.
+ */
+public class OidcProviderMetadataTest {
+    private OidcProviderMetadata oidcSecRealm;
+
+    @Before
+    public void setUp() {
+        oidcSecRealm = new OidcProviderMetadata();
+    }
+
+    @Test
+    public void testClaimsParameterSupported() throws Exception {
+        assertFalse(oidcSecRealm.getClaimsParameterSupported());
```

Comment:
The OIDC provider metadata gets created by sending a request to the OIDC provider's `/.well-known` endpoint and then using `JsonSerialization#readValue` (as seen here https://github.com/wildfly-security/wildfly-elytron/blob/2.x/http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcClientConfiguration.java#L273) to parse what we get back from the well-known endpoint into the `OidcProviderMetadata` object.

I'm wondering if we can update these tests to make use of `JsonSerialization#readValue` to more closely mimic what happens in the code.

To do that, I'm wondering if we can call `JsonSerialization#readValue` directly with a string value for the metadata and that way we can control what options are specified/not specified in our string.

To get a sample string value to work with, you could try setting up a Keycloak OpenID provider locally (following the steps from one of the OIDC blog posts), try to hit the well-known endpoint, e.g., http://localhost:<KEYCLOAK_PORT>/realms/myrealm/.well-known/openid-configuration to get sample OIDC provider metadata.

We can then create hardcoded provider metadata strings to be used for these tests to test the case where these boolean options are present and not present.

Does that make sense?

Reply:
done